src/lib/expression/evaluation/expression_evaluator.cpp

```diff
@@ -416,15 +405,8 @@ ExpressionEvaluator::_evaluate_is_null_expression<ExpressionEvaluator::Bool>(con
   return std::make_shared<ExpressionResult<ExpressionEvaluator::Bool>>(std::move(result_values));
 }
 
-template <typename Result>
```

Comment:
We had definitions for unspecialized templates in the code (that `Fail`ed). I am not sure if the compiler ignored them because we never tried to instantiate them. However, it's more clear and gets rid of some bloaty definitions.